jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverter.java

```diff
@@ -38,1082 +38,1258 @@
-      // timeStamp is optional
-      if (timDataObjectNode.get(TIME_STAMP) != null) {
-         timDataObjectNode.put(TIME_STAMP,
-               translateISOTimeStampToMinuteOfYear(timDataObjectNode.get(TIME_STAMP).asText()));
-      }
-
-      // urlB is optional but does not need replacement
-
-      // dataFrames are required
-      timDataObjectNode.set(DATA_FRAMES_STRING, transformDataFrames(timDataObjectNode.get(DATAFRAMES)));
-      timDataObjectNode.remove(DATAFRAMES);
-   }
-
-   public static ObjectNode transformDataFrames(JsonNode dataFrames) throws JsonUtilsException {
-
-      if (dataFrames == null) {
-         return JsonUtils.newNode();
-      }
-
-      ArrayNode replacedDataFrames = JsonUtils.newNode().arrayNode();
-      if (dataFrames.isArray()) {
-         Iterator<JsonNode> dataFramesIter = dataFrames.elements();
-
-         while (dataFramesIter.hasNext()) {
-            ObjectNode oldFrame = (ObjectNode) dataFramesIter.next();
-            replaceDataFrame(oldFrame);
-            // wrap each data frame inside a TravelerDataFrame
-            replacedDataFrames.add(oldFrame);
-         }
-      }
-
-      return JsonUtils.newObjectNode(TRAVELER_DATA_FRAME_STRING, replacedDataFrames);
-   }
-
-   public static void replaceDataFrame(ObjectNode dataFrame) throws JsonUtilsException {
-
-      // INPUT
-      //////
-      // "dataframes": [
-      //
-      // "startDateTime": "2017-08-02T22:25:00.000Z",
-      // "durationTime": 1,
-      // "frameType": "1",
-      // "sspTimRights": "0",
-      // "msgID": "RoadSignID",
-      // "position": {
-      // "latitude": "41.678473",
-      // "longitude": "-108.782775",
-      // "elevation": "917.1432"
-      // },
-      // "viewAngle": "1010101010101010",
-      // "mutcd": "5",
-      // "crc": "0000000000000000",
-      // "priority": "0",
-      // "sspLocationRights": "3",
-      // "regions": []
-      // "sspMsgTypes": "2",
-      // "sspMsgContent": "3",
-      // "content": "advisory",
-      // "items": [
-      // "513"
-      // ],
-      // "url": "null"
-      //
-      // ]
-
-      /// OUTPUT:
-      //////
-      // <dataFrames>
-      // <TravelerDataFrame>
-      // <startYear>2017</startYear>
-      // <startTime>308065</startTime>
-      // </TravelerDataFrame>
-      // </dataFrames>
-
-      // replace sspTimRights with notUsed=0 (to conform with J2735 2020)
-      dataFrame.put(NOT_USED, 0);
-      dataFrame.remove(SSP_TIM_RIGHTS);
-
-      // replace sspLocationRights with notUsed1=0 (to conform with J2735 2020)
-      dataFrame.put(NOT_USED_1, 0);
-      dataFrame.remove(SSP_LOCATION_RIGHTS);
-
-      // set frameType value
-      dataFrame.set("frameType", JsonUtils.newNode().put(dataFrame.get("frameType").asText(), EMPTY_FIELD_FLAG));
-
-      // replace sspMsgContent/sspMsgRights1 with notUsed2=0 (to conform with J2735 2020)
-      dataFrame.put(NOT_USED_2, 0);
-      dataFrame.remove(SSP_MSG_CONTENT);
-      dataFrame.remove(SSP_MSG_RIGHTS_1);

```

Comment:
Are the "do not use" fields now nullable so we're just not setting them?

Reply:
These fields are not nullable. The JSON is expected to conform to J2735-2024 standards before reaching this converter.